pkg/3scale/amp/component/apicast.go

```diff
@@ -843,13 +865,14 @@ func (apicast *Apicast) stagingPodAnnotations() map[string]string {
 	return annotations
 }
 
-func (apicast *Apicast) productionPodAnnotations() map[string]string {
+func (apicast *Apicast) productionPodAnnotations(watchedSecretAnnotations map[string]string) map[string]string {
 	annotations := map[string]string{
-		"prometheus.io/scrape": "true",
-		"prometheus.io/port":   "9421",
+		"prometheus.io/scrape":         "true",
+		"prometheus.io/port":           "9421",
+		APIcastEnvironmentCMAnnotation: apicast.envConfigMapHash(),
 	}
 
-	for key, val := range apicast.Options.ProductionAdditionalPodAnnotations {
```

Comment:
Did we allow to specify annotations via apim? 
If so, are we removing this functionality?

Reply:
Yes we do allow users to specify annotations that get injected into the Pod but not into the Deployment. The watched secret annotations get injected into the Deployment but not the Pod so there's no risk of them overwriting one another. So we can keep the existing functionality without changes.